### ISSUE TITLE
Round robin dns issues

### ISSUE THREAD

**Issue by fluential** [wazuh/wazuh#511]
Hi there,

I've noticed that ossec/wazuh is not able to correctly handle round robin dns entries.
I am running multiple wazuh managers behind AWS elastic load balancer. Ossec counters are disabled on server and client side - so every client could talk to any server. I am not using cluster funcitonality as the keys are distributed via different means.

When configuring client as follow:
```
  <server>
    <address>internal-xxxxx-zzz-8854157461.eu-west-1.elb.amazonaws.com</address>
    <protocol>tcp</protocol>
    <port>1514</port>
    </server>
  </client>
```
agent is staring I can see following:
```
8/04/04 15:41:40 ossec-agentd: INFO: Server IP Address: internal-xxxxx-zzz-8854157461.eu-west-1.elb.amazonaws.com/10.2.2.10
(...)
2018/04/04 15:43:03 ossec-agentd: INFO: Trying to connect to server (internal-xxxxx-zzz-8854157461.eu-west-1.elb.amazonaws.com/10.2.2.10:1514).
2018/04/04 15:43:25 ossec-agentd: WARNING: Unable to get hostname for 'internal-xxxxx-zzz-8854157461.eu-west-1.elb.amazonaws.com.
2018/04/04 15:43:25 ossec-agentd: INFO: Trying to connect to server (internal-xxxxx-zzz-8854157461.eu-west-1.elb.amazonaws.com/10.2.2.10:1514).
2018/04/04 15:43:25 ossec-agentd: WARNING: (4101): Waiting for server reply (not started). Tried: 'internal-xxxxx-zzz-8854157461.eu-west-1.elb.amazonaws.com'.
2018/04/04 15:43:42 ossec-agentd: WARNING: Unable to get hostname for 'internal-xxxxx-zzz-8854157461.eu-west-1.elb.amazonaws.com.
2018/04/04 15:43:42 ossec-agentd: INFO: Trying to connect to server (internal-xxxxx-zzz-8854157461.eu-west-1.elb.amazonaws.com/10.2.2.10:1514).
2018/04/04 15:44:04 ossec-agentd: WARNING: Unable to get hostname for 'internal-xxxxx-zzz-8854157461.eu-west-1.elb.amazonaws.com'.
```

```
# host internal-xxxxx-zzz-8854157461.eu-west-1.elb.amazonaws.com
internal-xxxxx-zzz-8854157461.eu-west-1.elb.amazonaws.comhas address 10.2.2.120
internal-xxxxx-zzz-8854157461.eu-west-1.elb.amazonaws.comhas address 10.2.2.10
# nc -w0 -v internal-xxxxx-zzz-8854157461.eu-west-1.elb.amazonaws.com 1514
Connection to internal-xxxxx-zzz-8854157461.eu-west-1.elb.amazonaws.com 1514 port [tcp/*] succeeded!
```

Everything is fine as long as round robin dns is NOT returned.
I would expect for ossec agent to correctly establish connection to a randomly returned ip address.

**Comment by fluential** [comment]
Just wanted to check if you need more info on that ticket?

**Comment by vikman90** [comment]
Hi @fluential,

sorry for the late response. Yes, we are aware of this issue: the agent performs a _chroot_ jail just after loading its settings. Due to this jail, the agent loses access to some files and libraries that it needs to resolve DNS names.

We have got a similar problem in the manager while implementing a file download feature and we addressed that issue with a solution that I think that could also solve this problem: using an external non-jailed service for file download and DNS resolving.

Code for agents is critical so we'd like to release the downloading module only for managers for the first version. Let me mark this issue as a bug and we will fix it in the next release.

Thank you for reporting.
Best regards.

**Comment by ewolkowicz** [comment]
I would like to note the same issue here.  I'm not using an ELB but I am dynamically updating the DNS when a new manager spins up so for now I will have to reserve an IP on a secondary interface that gets bound to the instance during bootstrap.  I would really love to see this feature this year before I move to put things behind a balancer when we need to scale up.

**Comment by vikman90** [comment]
Hi @fluential @ewolkowicz,

This issue is related to #2770 and should have been fixed by PR @4241. It will be available in the next minor release: 3.12.

Thank you for your feedback, and sorry to have taken so long to solve it.

Best regards.
